core/templates/services/playthrough-issues-backend-api.service.ts

```diff
@@ -15,87 +15,106 @@
-        if (cachedIssues !== null) {
-          return $q.resolve(cachedIssues);
-        } else {
-          return $http.get(getFullIssuesUrl(explorationId), {
-            params: {
-              exp_version: explorationVersion
-            }
-          }).then(function(response) {
-            var unresolvedIssueBackendDicts = response.data;
-            cachedIssues = unresolvedIssueBackendDicts.map(
-              PlaythroughIssueObjectFactory.createFromBackendDict);
-            return cachedIssues;
-          });
-        }
-      },
-      fetchPlaythrough: function(expId, playthroughId) {
-        return $http.get(getFullPlaythroughUrl(expId, playthroughId)).then(
-          function(response) {
-            var playthroughBackendDict = response.data;
-            return PlaythroughObjectFactory.createFromBackendDict(
-              playthroughBackendDict);
-          });
-      },
-      resolveIssue: function(issueToResolve, expId, expVersion) {
-        return $http.post(getFullResolveIssueUrl(expId), {
-          exp_issue_dict: issueToResolve.toBackendDict(),
-          exp_version: expVersion
-        }).then(function() {
-          var issueIndex = cachedIssues !== null ?
-            cachedIssues.findIndex(function(issue) {
-              return angular.equals(issue, issueToResolve);
-            }) : -1;
-          if (issueIndex === -1) {
-            var invalidIssueError = new Error(
-              'An issue which was not fetched from the backend has been ' +
-              'resolved');
-            return $q.reject(invalidIssueError);
-          } else {
-            cachedIssues.splice(issueIndex, 1);
-          }
-        });
-      },
-    };
-  }]);
+  private getFullIssuesUrl(explorationId: string): string {
+    return this.urlInterpolationService.interpolateUrl(
+      ServicesConstants.FETCH_ISSUES_URL, {
+        exploration_id: explorationId
+      });
+  }
+
+  private getFullPlaythroughUrl(expId: string, playthroughId: string): string {
+    return this.urlInterpolationService.interpolateUrl(
+      ServicesConstants.FETCH_PLAYTHROUGH_URL, {
+        exploration_id: expId,
+        playthrough_id: playthroughId
+      });
+  }
+
+  private getFullResolveIssueUrl(explorationId: string): string {
+    return this.urlInterpolationService.interpolateUrl(
+      ServicesConstants.RESOLVE_ISSUE_URL, {
+        exploration_id: explorationId
+      });
+  }
+
+  // TODO(#7165): This has been marked any since marking explorationVersion
+  // to number throws an error. "Type 'number' is not assignable to type
+  // 'string | string[]'" and if this is marked to string it throws an
+  // error "TS2345: Argument of type '1' is not assignable to parameter
+  // of type 'string'".
+  fetchIssues(explorationId: string, explorationVersion: any): Promise<string> {
+    if (this.cachedIssues !== null) {
+      return new Promise((resolve) => resolve(this.cachedIssues));
```

Comment:
Please prefer `$q` to `Promise`. Also, this can be simplified to:

```typescript
if (this.cachedIssues !== null) {
  return $q.resolve(this.cachedIssues);
}
```

Reply:
I referenced this example ->
https://github.com/oppia/oppia/pull/8427/files,
that's why changed it into a Promise, what is the correct approach?

Reply:
We should import the `$q` service, and use that to create Promise instances.

Reply:
Missed this by mistake, will complete this in next pass

Reply:
$q is something that works with AngularJS. For Angular we are using new Promise.

Reply:
Got it, in that case please use `Promise.resolve(value)` instead ([source](https://codecraft.tv/courses/angular/es6-typescript/promises/#_immediate_resolution_or_rejection)).

Reply:
This still isn't done? Please change to:

```typescript
if (this.cachedIssues !== null) {
  return Promise.resolve(this.cachedIssues);
}
```